### ISSUE TITLE
.gitignore and .npmignore conflict on publish

### ISSUE THREAD

**Issue by calebmer** [npm/npm#8796]
After upgrading to npm I discovered I couldn't publish my modules with a babel compile step. In my `.npmignore` file I include a `src` folder, in my `.gitignore` file I include a `lib` folder. For a pre-publish script I compile the ES6 code in `src` to ES5 code in `lib` with babel.

When publishing the compilation occurs, but then I get the following error: `npm ERR! Cannot read property 'main' of undefined`. This error only occurs in npm version 2.12.x and up, including npm version 3.x.x. I am not having any problems publishing in npm version 2.11.0

Here is the npm log:
```
0 info it worked if it ends with ok
1 verbose cli [ 'node',
1 verbose cli   '/Users/webdev/.nvm/versions/node/v0.12.5/bin/npm',
1 verbose cli   'publish' ]
2 info using npm@2.12.1
3 info using node@v0.12.5
4 verbose publish [ '.' ]
5 silly cache add args [ '.', null ]
6 verbose cache add spec .
7 silly cache add parsed spec { raw: '.',
7 silly cache add   scope: null,
7 silly cache add   name: null,
7 silly cache add   rawSpec: '.',
7 silly cache add   spec: '/Users/webdev/projects/calebmer/mongold',
7 silly cache add   type: 'directory' }
8 verbose addLocalDirectory /Users/webdev/.npm/mongold/0.5.8/package.tgz not in flight; packing
9 verbose tar pack [ '/Users/webdev/.npm/mongold/0.5.8/package.tgz',
9 verbose tar pack   '/Users/webdev/projects/calebmer/mongold' ]
10 verbose tarball /Users/webdev/.npm/mongold/0.5.8/package.tgz
11 verbose folder /Users/webdev/projects/calebmer/mongold
12 info prepublish mongold@0.5.8
13 verbose unsafe-perm in lifecycle true
14 verbose stack TypeError: Cannot read property 'main' of undefined
14 verbose stack     at Packer.applyIgnores (/Users/webdev/.nvm/versions/node/v0.12.5/lib/node_modules/npm/node_modules/fstream-npm/fstream-npm.js:108:30)
14 verbose stack     at Packer.<anonymous> (/Users/webdev/.nvm/versions/node/v0.12.5/lib/node_modules/npm/node_modules/fstream-npm/node_modules/fstream-ignore/ignore.js:61:15)
14 verbose stack     at Packer.emit (events.js:110:17)
14 verbose stack     at FileReader.<anonymous> (/Users/webdev/.nvm/versions/node/v0.12.5/lib/node_modules/npm/node_modules/fstream/lib/dir-reader.js:131:12)
14 verbose stack     at FileReader.emit (events.js:107:17)
14 verbose stack     at go (/Users/webdev/.nvm/versions/node/v0.12.5/lib/node_modules/npm/node_modules/fstream/lib/reader.js:216:12)
14 verbose stack     at go (/Users/webdev/.nvm/versions/node/v0.12.5/lib/node_modules/npm/node_modules/fstream/lib/reader.js:217:7)
14 verbose stack     at statCb (/Users/webdev/.nvm/versions/node/v0.12.5/lib/node_modules/npm/node_modules/fstream/lib/reader.js:218:7)
14 verbose stack     at FSReqWrap.oncomplete (evalmachine.<anonymous>:95:15)
15 verbose cwd /Users/webdev/projects/calebmer/mongold
16 error Darwin 14.3.0
17 error argv "node" "/Users/webdev/.nvm/versions/node/v0.12.5/bin/npm" "publish"
18 error node v0.12.5
19 error npm  v2.12.1
20 error Cannot read property 'main' of undefined
21 error If you need help, you may report this error at:
21 error     <https://github.com/npm/npm/issues>
22 verbose exit [ 1, true ]
```

**Comment by iarna** [comment]
Yup, looks like this was introduced w/ https://github.com/npm/fstream-npm/commit/183d071

**Comment by Goodmind** [comment]
how to fix this issue?

**Comment by othiym23** [comment]
Fixed by d338668601d1ebe5247a26237106e80ea8cd7f48 / #8858, which was included in `npm@2.13.1`. Thanks for reporting this, and your patience!
